From: Colin Watson <cjwatson@debian.org>
Date: Wed, 16 Oct 2024 16:34:53 +0100
Subject: test: Make git archive prefix fit in 32-bit ssize_t

On 32-bit architectures, the `git` integration tests fail with:

  TASK [git : ARCHIVE | Generate an archive prefix] ******************************
  An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OverflowError: Python int too large to convert to C ssize_t

It doesn't seem vital that this prefix should be larger than a signed
32-bit integer can hold, so drop one bit.

Forwarded: https://github.com/ansible/ansible/pull/84129
Last-Update: 2024-10-16
---
 test/integration/targets/git/tasks/archive.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/integration/targets/git/tasks/archive.yml b/test/integration/targets/git/tasks/archive.yml
index 588148d..bebe306 100644
--- a/test/integration/targets/git/tasks/archive.yml
+++ b/test/integration/targets/git/tasks/archive.yml
@@ -84,7 +84,7 @@
 
 - name: ARCHIVE | Generate an archive prefix
   set_fact:
-    git_archive_prefix: '{{ range(2 ** 31, 2 ** 32) | random }}' # Generate some random archive prefix
+    git_archive_prefix: '{{ range(2 ** 30, 2 ** 31) | random }}' # Generate some random archive prefix
 
 - name: ARCHIVE | Archive repo using various archival format and with an archive prefix
   git:
